src/cirrus/test.py

```diff
@@ -22,24 +22,59 @@ def build_parser(argslist):
     parser = ArgumentParser(
         description='git cirrus test command'
     )
-    parser.add_argument('test', nargs='+')
-    parser.add_argument('--suite') 
+    subparsers = parser.add_subparsers(dest='command')
+    test_command = subparsers.add_parser('test')
+    test_command.add_argument(
+        '--suite',
+        help='test suite configuration to use as defined in the test-<suite> section of cirrus.conf',
+        default='default'
+    )
+    test_command.add_argument(
+        '--mode',
+        choices=['nosetests', 'tox'],
+        default=None,
+        help='Choose test runner framework'
+    )
+    test_command.add_argument(
+        '--test-options',
+        default='',
+        dest='options',
+        help='Optional args to pass to test runner'
+    )
 
     opts = parser.parse_args(argslist)
     return opts
 
 
-@nottest
-def nose_test(location):
+def nose_run(config, opts):
     """
     _nose_test_
 
-    Locally activate vitrualenv and run tests
+    Locally activate vitrualenv and run tests via nose
     """
-    config = load_configuration()
-    local('. ./{0}/bin/activate && nosetests -w {1}'.format(
-        config.venv_name(),
-        config.test_where(location)))
+    where = config.test_where(opts.suite)
+    local(
+        '. ./{0}/bin/activate && nosetests -w {1} {2}'.format(
+            config.venv_name(),
+            where,
+            opts.options
+        )
+    )
+
+
+def tox_run(config, opts):
+    """
+    tox test
+
+    activate venv and run tox test suite
+
+    """
+    local(
+        '. ./{0}/bin/activate && tox {1}'.format(
+            config.venv_name(),
+            opts.options
```

Comment:
No, it just expects a top level tox.ini to run, I should probably check for that etc, but I think longer term this command may get plugins to wrap the different test environments, and support a richer API